### PR TITLE
fix: jwt subject is user identity

### PR DESCRIPTION
**Pull request by kuoche1712003**
## Why need this change? / Root cause: 
- Fix JWT subject misused.
## Changes made:
- Add findByIdentity method.
- Usecase use identity to find user.
- Test user add identity.
## Test Scope / Change impact:
- Room controller.
- Room usercases.
- RoomControllerTest.
## Issue
- resolves https://github.com/Game-as-a-Service/Lobby-Platform-Service/issues/125

### CODE REVIEW

File: application/src/main/kotlin/tw/waterballsa/gaas/application/usecases/KickPlayerUsecase.kt [hunk 1]
```diff
@@ -1,25 +1,31 @@
...
 @Named
 class KickPlayerUsecase(
     private val roomRepository: RoomRepository,
+    private val userRepository: UserRepository,
 ) {
     fun execute(request: Request) {
         with(request) {
             val room = roomRepository.findById(Room.Id(roomId)) ?: throw notFound(Room::class).message()
-            room.kickPlayer(Room.Player.Id(hostId), Room.Player.Id(playerId))
+            val hostPlayer = userRepository.findByIdentity(hostIdentity)?.toRoomPlayer()
```
Comment (Wally5077) [inline application/src/main/kotlin/tw/waterballsa/gaas/application/usecases/KickPlayerUsecase.kt:19]: `host`

File: application/src/main/kotlin/tw/waterballsa/gaas/application/usecases/KickPlayerUsecase.kt [hunk 2]
```diff
@@ -1,25 +1,31 @@
...
+                ?: throw notFound(User::class).message()
+            room.kickPlayer(hostPlayer.id, Room.Player.Id(playerId))
             roomRepository.update(room)
         }
     }
 
     data class Request(
         val roomId: String,
-        val hostId: String,
+        val hostIdentity: String,
```
Comment (Wally5077) [inline application/src/main/kotlin/tw/waterballsa/gaas/application/usecases/KickPlayerUsecase.kt:28]: `userIdentity`

File: application/src/main/kotlin/tw/waterballsa/gaas/application/usecases/LeaveRoomUsecase.kt [hunk 3]
```diff
@@ -25,6 +29,10 @@ class LeaveRoomUsecase(
         roomRepository.findById(roomId)
             ?: throw notFound(Room::class).id(roomId)
 
+    private fun findPlayerByIdentity(identity: String) =
+        userRepository.findByIdentity(identity)?.toRoomPlayer()
+            ?: throw notFound(User::class).message()
```
Comment (Wally5077) [inline application/src/main/kotlin/tw/waterballsa/gaas/application/usecases/LeaveRoomUsecase.kt:34]: ditto 換行

File: spring/src/test/kotlin/tw/waterballsa/gaas/spring/it/controllers/RoomControllerTest.kt [hunk 4]
```diff
@@ -520,7 +570,7 @@ class RoomControllerTest @Autowired constructor(
     private fun findRoomById(roomId: Room.Id): Room? =
         roomRepository.findById(roomId)
 
-    private fun ResultActions.thenPlayersShouldBeInTheRoom(roomId: Room.Id, vararg users: User) =
+    private fun thenPlayersShouldBeInTheRoom(roomId: Room.Id, vararg users: User) =
```
Comment (Wally5077) [inline spring/src/test/kotlin/tw/waterballsa/gaas/spring/it/controllers/RoomControllerTest.kt:573]: Ｑ：請說一下調整這裡的原因~

**Review by Wally5077 — CHANGES_REQUESTED** [review]
**Review done.**

File: spring/src/test/kotlin/tw/waterballsa/gaas/spring/it/controllers/RoomControllerTest.kt (hunk 4 above)
Comment (kuoche1712003) [reply spring/src/test/kotlin/tw/waterballsa/gaas/spring/it/controllers/RoomControllerTest.kt:573]: 我看 IDE 是灰字就順手把他拿掉了
還是要維持原樣？

**Review by Wally5077 — APPROVED** [review]
**LGTM**

**Pull request merged**